src/client/runner/automation/playback/select/index.js

```diff
@@ -0,0 +1,218 @@
+            this.eventState.mousedownPrevented = e.defaultPrevented;
+            eventUtils.preventDefault(e);
+            eventUtils.unbind(this.element, 'mousedown', onmousedown);
+        };
+
+        eventUtils.bind(this.element, 'mousedown', onmousedown);
+    }
+
+    _calculateAbsoluteStartPoint () {
+        var point = null;
+
+        if (!this.startNode)
+            point = selectUtils.getSelectionCoordinatesByPosition(this.element, this.startPos);
+        else if (this.backward)
+            point = selectUtils.getSelectionCoordinatesByNodeAndOffset(this.element, this.endNode, this.endOffset);
+        else
+            point = selectUtils.getSelectionCoordinatesByNodeAndOffset(this.element, this.startNode, this.startOffset);
+
+        return point || positionUtils.findCenter(this.element);
+    }
+
+    _calculateAbsoluteEndPoint () {
+        var point = null;
+
+        if (!this.startNode)
+            point = selectUtils.getSelectionCoordinatesByPosition(this.element, this.endPos);
+        else if (this.backward)
+            point = selectUtils.getSelectionCoordinatesByNodeAndOffset(this.element, this.startNode, this.startOffset);
+        else
+            point = selectUtils.getSelectionCoordinatesByNodeAndOffset(this.element, this.endNode, this.endOffset);
+
+        if (point)
+            return point;
+
+        // NOTE: if selection ends on an invisible symbol, we should try to find the last visible selection position
+        if (domUtils.isContentEditableElement(this.element) && this.startPos !== void 0)
+            return selectUtils.getLastVisibleSelectionPosition(this.element, this.startPos, this.endPos);
+
+        return positionUtils.findCenter(this.element);
+    }
+
+    _moveToPoint (point) {
+        selectUtils.scrollEditableElementByPoint(this.element, point);
+
+        this.clientPoint = selectUtils.excludeElementScroll(this.element, point);
+
+        var moveArguments = {
+            element: document.documentElement,
+            offsetX: this.clientPoint.x,
+            offsetY: this.clientPoint.y
+        };
+
+        return this._move(moveArguments);
+    }
+
+    _mousedown () {
+        return cursor
+            .leftButtonDown()
+            .then(() => {
+                this.eventArgs = SelectAutomation._calculateEventArguments(this.clientPoint);
+
+                // NOTE: in WebKit and IE, the mousedown event opens the select element's dropdown.
+                // Therefore, we should prevent mousedown and hide the dropdown (B236416)
+                var needCloseSelectDropDown = (browserUtils.isWebKit || browserUtils.isIE) &&
+                                              domUtils.isSelectElement(this.element);
+
+                if (needCloseSelectDropDown)
+                    this._bindMousedownHandler();
+
+                if (browserUtils.hasTouchEvents) {
+                    this.eventState.simulateDefaultBehavior = eventSimulator.touchstart(this.eventArgs.element,
```

Comment:
I propose to define `this.mousedownEvent = browserUtils.hasTouchEvents ? 'touchstart' : 'mousedown'` in the `constructor` to avoid condition blocks and make code more clean here. (the same for `mouseup`)